File: KantechGatewayApp/Jobs/IJobProcessor.cs
using System;

namespace KantechGatewayApp.KantechGatewayApp.Jobs
{
    public interface IJobProcessor
    {
        string Key { get; }
        string ScheduleName { get; }
        DateTime LastRunUtc { get; }
        string LastResult { get; }
        string Status { get; }

        void Configure(string jobKey);
        void MaybeRunBySchedule();
        void RunOnce();
    }
}

File: KantechGatewayApp/Jobs/IJobProcessor.vb
Namespace KantechGatewayApp.Jobs
    Public Interface IJobProcessor
        ReadOnly Property Key As String
        ReadOnly Property ScheduleName As String
        ReadOnly Property LastRunUtc As Date
        ReadOnly Property LastResult As String
        ReadOnly Property Status As String

        Sub Configure(jobKey As String)
        Sub MaybeRunBySchedule()
        Sub RunOnce()
    End Interface
End Namespace